core/container.go

```diff
@@ -1214,6 +1215,7 @@ func (container *Container) Publish(
 	platformVariants []*Container,
 	forcedCompression ImageLayerCompression,
 	mediaTypes ImageMediaTypes,
+	rewriteTimestamp dagql.Boolean,
```

Comment:
I think the arg here should *ideally* be a timestamp? Then, if left empty, we would default to `clientMetadata.SourceDateEpoch`.

Just because currently, to use this feature, you have to update your module code *and* have the client pass `SOURCE_DATE_EPOCH` which feels tricky - is there ever a reason why we wouldn't want to respect `SOURCE_DATE_EPOCH` if it was set by the client?

We're not tied strictly to the buildkit semantics here, so we're free to build the API we want :tada:

cc @sipsma in case you have opinions about this